centos2ol.sh

```diff
@@ -220,345 +290,384 @@ You may want select No to stop and raise an issue on ${github_url} for advice."
-		gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
-		gpgcheck=1
-		enabled=1
-
-		[ol8_appstream]
-		name=Oracle Linux 8 Application Stream ($basearch)
-		baseurl=https://yum.oracle.com/repo/OracleLinux/OL8/appstream/$basearch/
-		gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
-		gpgcheck=1
-		enabled=1
-
-		[ol8_UEKR6]
-		name=Latest Unbreakable Enterprise Kernel Release 6 for Oracle Linux $releasever ($basearch)
-		baseurl=https://yum.oracle.com/repo/OracleLinux/OL8/UEKR6/$basearch/
-		gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
-		gpgcheck=1
-		enabled=1
+step_info $LINENO "Download the content for 6 and 7 based systems and directly enter the content for 8 based systems"
+if (( STEPNO >= STARTNO  )) ; then
+   case "$os_version" in
+       8*)
+           cat > "switch-to-oraclelinux.repo" <<-'EOF'
+[ol8_baseos_latest]
+name=Oracle Linux 8 BaseOS Latest ($basearch)
+baseurl=https://yum.oracle.com/repo/OracleLinux/OL8/baseos/latest/$basearch/
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
+gpgcheck=1
+enabled=1
+
+[ol8_appstream]
+name=Oracle Linux 8 Application Stream ($basearch)
+baseurl=https://yum.oracle.com/repo/OracleLinux/OL8/appstream/$basearch/
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
+gpgcheck=1
+enabled=1
+
+[ol8_UEKR6]
+name=Latest Unbreakable Enterprise Kernel Release 6 for Oracle Linux $releasever ($basearch)
+baseurl=https://yum.oracle.com/repo/OracleLinux/OL8/UEKR6/$basearch/
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
+gpgcheck=1
+enabled=1
 EOF
-        ;;
-    *)
-        echo "Downloading Oracle Linux yum repository file..."
-        if ! curl -o "switch-to-oraclelinux.repo" "${yum_url}/${repo_file}"; then
-            exit_message "Could not download $repo_file from $yum_url.
-        Are you behind a proxy? If so, make sure the 'http_proxy' environment
-        variable is set with your proxy address."
-        fi
-        ;;
-esac
-
-
-echo "Looking for yumdownloader..."
-if ! have_program yumdownloader; then
-    # CentOS 6 mirrors are now offline, if yumdownloader tool is not present then
-    #  use OL6 as source for yum-utils and disable all other repos.
-    # Use the existing distributions copy for other platforms
-    case "$os_version" in
-        6*)
-            curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-oracle https://yum.oracle.com/RPM-GPG-KEY-oracle-ol6
-            yum -y install yum-utils --disablerepo \* --enablerepo ol6_latest || true
-            ;;
-        *)
-            yum -y install yum-utils --disablerepo ol\* || true
-            ;;
-    esac
-    dep_check yumdownloader
+           ;;
+       *)
+           echo "Downloading Oracle Linux yum repository file..."
+           if ! curl -o "switch-to-oraclelinux.repo" "${yum_url}/${repo_file}"; then
+               abend $LINENO "Could not download $repo_file from $yum_url.
+           Are you behind a proxy? If so, make sure the 'http_proxy' environment
```

Comment:
This should be the `https_proxy` environment variable, though I notice that's an issue with the existing version as well.